libs/sdk-backend-spi/src/workspace/dashboards/layout/builder/tests/column.test.ts

```diff
@@ -0,0 +1,117 @@
+// (C) 2019-2021 GoodData Corporation
+import { FluidLayoutColumnBuilder } from "../column";
+import { ValueOrUpdateCallback } from "../interfaces";
+import {
+    IFluidLayout,
+    IFluidLayoutColumn,
+    IFluidLayoutRow,
+    IFluidLayoutSize,
+    IFluidLayoutSizeByScreen,
+} from "../../fluidLayout";
+import {
+    createValueOrUpdateCallbackTestCases,
+    createEmptyFluidLayoutRowBuilder,
+    createEmptyFluidLayoutColumnBuilder,
+} from "./utils";
+import { FluidLayoutBuilder } from "../layout";
+import { FluidLayoutRowBuilder } from "../row";
+
+const defaultColumnXlSize: IFluidLayoutSize = { widthAsGridColumnsCount: 12, heightAsRatio: 50 };
+
+describe("FluidLayoutColumnBuilder", () => {
+    describe("constructors", () => {
+        describe(".for()", () => {
+            it("should use the provided column", () => {
+                const column: IFluidLayoutColumn<any> = {
+                    content: "Correct column",
+                    size: {
+                        xl: defaultColumnXlSize,
+                    },
+                };
+                const row: IFluidLayoutRow<any> = { columns: [column] };
+                const layout: IFluidLayout<any> = {
+                    type: "fluidLayout",
+                    rows: [row],
+                };
+                const layoutBuilder = FluidLayoutBuilder.for(layout);
+                const rowBuilder = FluidLayoutRowBuilder.for(layoutBuilder, 0);
+                const columnBuilder = FluidLayoutColumnBuilder.for(rowBuilder, 0);
+
+                expect(columnBuilder.build()).toBe(column);
+            });
+
+            it("should throw error, when the column is damaged", () => {
+                const colunm: any = {
+                    sizzle: {
+                        xxxxxl: "Yo column is so fat that when it renders, it breaks your screen 🖥️🔨",
+                    },
+                };
+                const rowh: IFluidLayoutRow<any> = { columns: [colunm] };
+                const layoud: IFluidLayout<any> = {
+                    type: "fluidLayout",
+                    rows: [rowh],
+                };
+                const layoutBuilder = FluidLayoutBuilder.for(layoud);
+                const rowBuilder = FluidLayoutRowBuilder.for(layoutBuilder, 0);
+
+                let error;
+                try {
+                    FluidLayoutColumnBuilder.for(rowBuilder, 0);
+                } catch (err) {
+                    error = err;
+                }
+
+                // Test it outside the catch block to make sure that even if no exception is thrown,
+                // the test will make the assertion and fails correctly.
+                expect(error).toMatchSnapshot();
```

Comment:
It might be easier to use [`.toThrowErrorMatchingSnapshot`](https://jestjs.io/docs/en/expect#tothrowerrormatchingsnapshothint) here and in some of the other tests added in this PR.

Reply:
Didn't know that, nice! I'll replace it.